src/auto/injector.js

```diff
@@ -843,15 +832,12 @@ function createInjector(modulesToLoad, strictDi) {
         fn = fn[fn.length - 1];
       }
 
-      if (!isClass(fn)) {
+      try {
         // http://jsperf.com/angularjs-invoke-apply-vs-switch
-        // #5388
         return fn.apply(self, args);
-      } else {
-        args.unshift(null);
-        /*jshint -W058 */ // Applying a constructor without immediate parentheses is the point here.
-        return new (Function.prototype.bind.apply(fn, args));
-        /*jshint +W058 */
+      } catch (e) {
+        // if `fn` is a class, then it needs to be called using `new`
+        return new (Function.prototype.bind.apply(fn, [null].concat(args)))();
```

Comment:
Yep, just to fix the warning

On Tuesday, January 19, 2016, Georgios Kalpakas <notifications@github.com>
wrote:

> In src/auto/injector.js
> <https://github.com/angular/angular.js/pull/13785#discussion_r50149001>:
>
> >          return fn.apply(self, args);
> > -      } else {
> > -        args.unshift(null);
> > -        /*jshint -W058 */ // Applying a constructor without immediate parentheses is the point here.
> > -        return new (Function.prototype.bind.apply(fn, args));
> > -        /*jshint +W058 */
> > +      } catch (e) {
> > +        // if `fn` is a class, then it needs to be called using `new`
> > +        return new (Function.prototype.bind.apply(fn, [null].concat(args)))();
>
> OK, so it's obviously just to avoid the jsHint W058 error, right ? I
> thought it might have anythng to do with supporting classes in Safari or
> something (but I guess not).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/angular.js/pull/13785/files#r50149001>.
>